drivers/timer/stm32_lptim_timer.c

```diff
@@ -281,44 +291,26 @@ static int sys_clock_driver_init(const struct device *dev)
 	LL_SRDAMR_GRP1_EnableAutonomousClock(LL_SRDAMR_GRP1_PERIPH_LPTIM1AMEN);
 #endif
 
-#if defined(CONFIG_STM32_LPTIM_CLOCK_LSI)
-	/* enable LSI clock */
-#ifdef CONFIG_SOC_SERIES_STM32WBX
-	LL_RCC_LSI1_Enable();
-	while (!LL_RCC_LSI1_IsReady()) {
-#else
-	LL_RCC_LSI_Enable();
-	while (!LL_RCC_LSI_IsReady()) {
-#endif /* CONFIG_SOC_SERIES_STM32WBX */
-		/* Wait for LSI ready */
-	}
-
-	LL_RCC_SetLPTIMClockSource(LL_RCC_LPTIM1_CLKSOURCE_LSI);
+	/* Enable LPTIM clock source */
+	clock_control_configure(clk_ctrl, (clock_control_subsys_t *) &lptim_clk[1],
+				NULL);
 
-#else /* CONFIG_STM32_LPTIM_CLOCK_LSI */
-#if defined(LL_APB1_GRP1_PERIPH_PWR)
-	/* Enable the power interface clock */
-	LL_APB1_GRP1_EnableClock(LL_APB1_GRP1_PERIPH_PWR);
-#endif /* LL_APB1_GRP1_PERIPH_PWR */
+	/* Get LPTIM clock freq */
+	clock_control_get_rate(clk_ctrl, (clock_control_subsys_t *) &lptim_clk[1],
+			       &lptim_clock_freq);
 
-	/* enable backup domain */
-	LL_PWR_EnableBkUpAccess();
+	/* Set LPTIM time base based on clck source freq */
 
-	/* enable LSE clock */
-	LL_RCC_LSE_DisableBypass();
-#ifdef RCC_BDCR_LSEDRV_Pos
-	LL_RCC_LSE_SetDriveCapability(STM32_LSE_DRIVING << RCC_BDCR_LSEDRV_Pos);
-#endif
-	LL_RCC_LSE_Enable();
-	while (!LL_RCC_LSE_IsReady()) {
-		/* Wait for LSE ready */
+	/* A 32bit value cannot exceed 0xFFFFFFFF/lptim_time_base counting cycles.
+	 * This is for example about of 65000 x 2000ms when clocked by LSI
```

Comment:
The original comment should not be moved,
And to explain the 0xFA00 value:
`When clocked by LSI @32K:   32000Hz x 2s = 65000 = 0xFA00`